http-client/src/main/java/com/facebook/airlift/http/client/thrift/ThriftResponseHandler.java

```diff
@@ -49,6 +53,23 @@ public ThriftResponseHandler(ThriftCodec<T> thriftCodec)
     @Override
     public ThriftResponse<T> handle(Request request, Response response)
     {
+        if (isClientError(response.getStatusCode())) {
+            ByteSource byteSource = new ByteSource() {
```

Comment:
For readability I'd factor this into a private method.

Reply:
done